engines/testbed/image.cpp

```diff
@@ -0,0 +1,346 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/file.h"
+
+#include "graphics/managed_surface.h"
+#include "graphics/paletteman.h"
+#include "graphics/pm5544.h"
+#include "graphics/screen.h"
+#include "image/bmp.h"
+#include "image/gif.h"
+#include "image/iff.h"
+#include "image/jpeg.h"
+#include "image/pcx.h"
+#include "image/pict.h"
+#include "image/png.h"
+#include "image/tga.h"
+#include "image/xbm.h"
+
+#include "testbed/image.h"
+#include "testbed/testsuite.h"
+
+namespace Testbed {
+
+namespace ImageTests {
+TestExitStatus testRenderPM5544();
+TestExitStatus testBitmapDecoder();
+TestExitStatus testJPEGDecoder();
+TestExitStatus testGIFDecoder();
+TestExitStatus testPCXDecoder();
+TestExitStatus testPICTDecoder();
+TestExitStatus testPNGDecoder();
+TestExitStatus testTGADecoder();
+TestExitStatus testXBMDecoder();
+bool testImageDecoder(Common::Path &filepath, Image::ImageDecoder &decoder);
+}
+
+ImageTestSuite::ImageTestSuite() {
+	// Add tests here
+
+	// Render base image
+	addTest("testRenderPM5544", &ImageTests::testRenderPM5544);
+
+	// Test image decoders
+	addTest("testBitmapDecoder", &ImageTests::testBitmapDecoder);
+	addTest("testJPEGDecoder", &ImageTests::testJPEGDecoder);
+	addTest("testGIFDecoder", &ImageTests::testGIFDecoder);
+	addTest("testPCXDecoder", &ImageTests::testPCXDecoder);
+	addTest("testPICTDecoder", &ImageTests::testPICTDecoder);
+	addTest("testPNGDecoder", &ImageTests::testPNGDecoder);
+	addTest("testTGADecoder", &ImageTests::testTGADecoder);
+
+	// External XBM files are not yet supported
+	//addTest("testXBMDecoder", &ImageTests::testXBMDecoder);
+}
+
+TestExitStatus ImageTests::testRenderPM5544() {
+	Common::String info = "Render test pattern?\n"
+						  "This test pattern is the comparison reference for the other image decoders.";
+
+	if (Testsuite::handleInteractiveInput(info, "OK", "Skip", kOptionRight)) {
+		Testsuite::logPrintf("Info! Skipping test : testRenderPM5544()\n");
+		return kTestSkipped;
+	}
+
+	int xres = 320, yres = 240;
+	g_system->beginGFXTransaction();
+	g_system->initSize(xres, yres);
+	g_system->endGFXTransaction();
+
+	Graphics::ManagedSurface *pm5544 = Graphics::renderPM5544(xres, yres);
+	// Clear the version string
+	pm5544->fillRect(Common::Rect(112, 176, 208, 192), 0);
+
+	byte palette[768];
+	pm5544->grabPalette(palette, 0, 256);
+	// Common::Path filename("image/pm5544.bmp");
+	// Common::DumpFile dumpFile;
+	// bool result = dumpFile.open(filename);
+	// if (result) {
+	// 	result = Image::writeBMP(dumpFile, pm5544->rawSurface(), palette);
+	// }
```

Comment:
Leftover debug code?

```suggestion
```

Reply:
Used to generate the base image for the magick script. I left to potentially regenerate, but we can remove if desired.

Reply:
Thanks